airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/InputConsumerTask.kt

```diff
@@ -78,13 +78,14 @@ class DefaultInputConsumerTask(
                     )
                 queue.publish(reserved.replace(wrapped))
             }
-            is DestinationStreamComplete,
-            is DestinationStreamIncomplete -> {
+            is DestinationStreamComplete -> {
                 reserved.release() // safe because multiple calls conflate
                 val wrapped = StreamCompleteWrapped(index = manager.markEndOfStream())
                 queue.publish(reserved.replace(wrapped))
                 queue.close()
             }
+            is DestinationStreamIncomplete ->
+                throw IllegalStateException("Stream $stream failed upstream, cannot continue.")
```

Comment:
we might want to define a custom exception class for this - warehouse destinations will likely want to catch it and handle specific cases themselves (and then propagate the exception upward)

Reply:
Interesting. What's the use case? Would that just apply to this exception or any stream failure?

Reply:
Interesting. Okay, when the time comes that can be accomplished by
A) adding that exception here
B) handling it during `streamLoader::close(failureResult)`